.pre-commit-config.yaml

```diff
@@ -0,0 +1,31 @@
+repos:
+    -   repo: https://github.com/pycqa/isort
+        rev: 5.8.0
+        hooks:
+        -    id: isort
+             additional_dependencies: ["colorama>=0.4.3"]
+
+    -   repo: https://github.com/psf/black
+        rev: 21.4b0
+        hooks:
+        -   id: black
+
+    -   repo: https://gitlab.com/pycqa/flake8.git
+        rev: 3.8.0
+        hooks:
+        -   id: flake8
+
+    -   repo: https://github.com/pycqa/pydocstyle
+        rev: 6.1.0
+        hooks:
+        -   id: pydocstyle
+            exclude: (tests|docs)
```

Comment:
Is this actually needed? I assume it still uses the settings in `pyproject.toml` doesn't it?

Reply:
`pre-commit` does not use `pyproject.toml` so it is required. Ref: [here](https://github.com/pre-commit/pre-commit/issues/1165)

Reply:
Wow, that developer is... not friendly. And this is an extremely obvious feature to have with no downsides whatsoever. Are you sure we actually want to use pre-commit?

Reply:
I don't have strong opinions about us using it. For me `pre-commit` was a convenient tool to catch style and formatting errors on my side rather than push it to the CI workflow and find out about it later. My original goal was to provide this option should any current/future contributors feel the same way and **if** it's harmless to maintain, add it to the repo.

As I mentioned earlier the one con to `pre-commit` is that any major changes to `pyproject.toml` will need to be reflected in the `pre-commit` config. 

Updated black version? Need to do that here. Added a new dependency that supports type hints? Need to add that here.
To be fair, this can be caught pretty easily but yeah this is something to be considered.

Truth be told, if the final call is to close this PR and refer users to it when new contributors keep hitting style/formatting errors that honestly fine too.